Assets/MixedRealityToolkit.Providers/WindowsMixedReality/WindowsMixedRealitySpatialMeshObserver.cs

```diff
@@ -422,7 +452,18 @@ private void RequestMesh(SurfaceId surfaceId)
             {
                 newMesh = SpatialAwarenessMeshObject.Create(null, MeshPhysicsLayer, meshName, surfaceId.handle);
 
-                worldAnchor = newMesh.GameObject.AddComponent<WorldAnchor>();
+                // The WorldAnchor component places its object where the anchor is in the same space as the camera. 
+                // But since the camera is repositioned by the MixedRealityPlayspace's transform, the meshes' transforms
+                // should also the WorldAnchor position repositioned by the MixedRealityPlayspace's transform.
+                // So rather than put the WorldAnchor on the mesh's GameObject, the WorldAnchor is placed out of the way in the scene,
+                // and its transform is concatenated with the Playspace transform to compute the transform on the mesh's object.
+                // That adapting the WorldAnchor's transform into playspace is done by the internal PlayspaceAdapter component.
+                // The GameObject the WorldAnchor is placed on is unimportant, but it is convenient for cleanup to make it a child
+                // of the GameObject whose transform will track it.
+                GameObject anchorHolder = new GameObject(meshName + "_anchor");
```

Comment:
Do we ever clean up this GameObject when the spatial object is recycled?

Reply:
The anchorHolder is attached as a child node to the newMesh.GameObject. It is cleaned up as part of the newMesh.GameObject tree.